framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -10399,5 +10608,143 @@ void VulkanReplayConsumerBase::SetCurrentBlockIndex(uint64_t block_index)
     main_thread_queue_.poll();
 }
 
+void VulkanReplayConsumerBase::LoadPipelineCache(format::HandleId id, std::vector<char>& pipelineCacheData)
+{
+    FILE*   file  = nullptr;
+    int32_t error = util::platform::FileOpen(&file, options_.load_pipeline_cache_filename.c_str(), "r");
+    if (error)
+    {
+        GFXRECON_LOG_ERROR("Could not open pipeline cache file '%s' for loading. Error: '%s'",
+                           options_.save_pipeline_cache_filename.c_str(),
+                           strerror(error));
+        return;
+    }
+
+    format::HandleId idRead;
+    uint64_t         cacheSizeRead;
+
```

Comment:
Should we at least add a header so we know we're reading a pipeline cache file?

Reply:
hmm, not sure. the feature will only read files explicitly passed on cmd-line. even when loading 'valid' pipeline-cache-files written by gfxr, they might still be in-compatible for other reasons (different driver/vendor). while I also think this file could contain 'some' more useful metadata, like a header identifying it as pipeline-cache, meta-data about driver/vendor producing the cache etc. - I don't see an immediate danger or risk of leaving the plain cache-file structure as-is and improve on it in a follow-up, when we gathered some experience using it. what do you think?